ionoscloud/resource_autoscaling_group_test.go

```diff
@@ -0,0 +1,563 @@
+					resource.TestCheckResourceAttr(resourceAGName, "max_replica_count", "5"),
+					resource.TestCheckResourceAttr(resourceAGName, "min_replica_count", "1"),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.metric", "INSTANCE_CPU_UTILIZATION_AVERAGE"),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.range", "PT24H"),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.scale_in_action.0.amount", "1"),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.scale_in_action.0.amount_type", string(autoscaling.ACTIONAMOUNT_ABSOLUTE)),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.scale_in_action.0.termination_policy_type", string(autoscaling.TERMINATIONPOLICYTYPE_OLDEST_SERVER_FIRST)),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.scale_in_action.0.cooldown_period", "PT5M"),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.scale_in_threshold", "33"),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.scale_out_action.0.amount", "1"),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.scale_out_action.0.amount_type", string(autoscaling.ACTIONAMOUNT_ABSOLUTE)),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.scale_out_action.0.cooldown_period", "PT5M"),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.scale_out_threshold", "77"),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.unit", "PER_HOUR"),
+					resource.TestCheckResourceAttr(resourceAGName, "replica_configuration.0.availability_zone", "AUTO"),
+					resource.TestCheckResourceAttr(resourceAGName, "replica_configuration.0.cores", "2"),
+					resource.TestCheckResourceAttr(resourceAGName, "replica_configuration.0.cpu_family", string(autoscaling.CPUFAMILY_INTEL_SKYLAKE)),
+					resource.TestCheckResourceAttrPair(resourceAGName, "replica_configuration.0.nic.0.lan", constant.LanResource+".autoscaling_lan_1", "id"),
+					resource.TestCheckResourceAttr(resourceAGName, "replica_configuration.0.nic.0.name", "nic_1"),
+					resource.TestCheckResourceAttr(resourceAGName, "replica_configuration.0.nic.0.dhcp", "true"),
+					resource.TestCheckResourceAttr(resourceAGName, "replica_configuration.0.ram", "2048"),
+					resource.TestCheckResourceAttr(resourceAGName, "replica_configuration.0.volume.0.image_alias", "ubuntu:latest"),
+					resource.TestCheckResourceAttr(resourceAGName, "replica_configuration.0.volume.0.name", "volume_1"),
+					resource.TestCheckResourceAttr(resourceAGName, "replica_configuration.0.volume.0.size", "30"),
+					utils.TestNotEmptySlice(constant.AutoscalingGroupResource, "replica_configuration.0.volume.0.ssh_keys"),
+					resource.TestCheckResourceAttr(resourceAGName, "replica_configuration.0.volume.0.type", "HDD"),
+					resource.TestCheckResourceAttr(resourceAGName, "replica_configuration.0.volume.0.boot_order", "AUTO"),
+					resource.TestCheckResourceAttr(resourceAGName, "replica_configuration.0.volume.0.bus", "IDE"),
```

Comment:
@cristiGuranIonos checks for `user_data` and `image_password` are missing but I think this is fine since we don't keep the password in the state, but not sure for the `user_data` tho, we need to double check this.